sockets/1.2/modules/ROOT/pages/edits/sockets-connector-examples.adoc

```diff
@@ -0,0 +1,263 @@
+
+[source,xml,linenums]
+----
+	<sockets:listener-config name="Sockets_Listener_config" doc:name="Sockets Listener config" >
+		<sockets:tcp-listener-connection host="localhost" port="8082" >
+			<sockets:protocol >
+				<sockets:direct-protocol />
+			</sockets:protocol>
+		</sockets:tcp-listener-connection>
+	</sockets:listener-config>
+
+	<sockets:listener-config name="Sockets_Listener_config1" doc:name="Sockets Listener config" >
+		<sockets:tcp-listener-connection host="localhost" port="8085" >
+			<sockets:protocol >
+				<sockets:direct-protocol />
+			</sockets:protocol>
+		</sockets:tcp-listener-connection>
+	</sockets:listener-config>
+
+	<sockets:request-config name="Sockets_Request_config" doc:name="Sockets Request config" >
+		<sockets:tcp-requester-connection host="localhost" port="8082" >
+			<sockets:protocol >
+				<sockets:direct-protocol />
+			</sockets:protocol>
+		</sockets:tcp-requester-connection>
+	</sockets:request-config>
+
+	<db:config name="Database_Config" doc:name="Database Config" >
+		<db:oracle-connection host="localhost" user="sys as sysdba" password="Oradoc_db1" serviceName="orclpdb1.localdomain" />
+	</db:config>
+
+	<file:config name="File_Config" doc:name="File Config" >
+		<file:connection />
+	</file:config>
+
+	<flow name="socket-testFlow">
+		<sockets:listener doc:name="Listener" config-ref="Sockets_Listener_config" outputMimeType="application/json" />
+		<db:select doc:name="Select" config-ref="Database_Config">
+			<db:sql >#["SELECT first_name, last_name from SYSTEM.MUSICIANS where id = :id"]</db:sql>
+			<db:input-parameters><![CDATA[#[{ 'id': payload.id }]]]></db:input-parameters>
+		</db:select>
+		<ee:transform doc:name="Transform Message">
+			<ee:message >
+				<ee:set-payload><![CDATA[%dw 2.0
+					output application/json --- 'name': payload[0].FIRST_NAME as String ++ ' ' ++ payload[0].LAST_NAME as String]]>
+				</ee:set-payload>
+			</ee:message>
+		</ee:transform>
+	</flow>
+
+	<flow name="socketSendAndReceive" >
+		<sockets:listener doc:name="Listener" config-ref="Sockets_Listener_config1" outputMimeType="application/json"/>
+		<sockets:send-and-receive doc:name="Send and receive" config-ref="Sockets_Request_config" outputMimeType="application/json">
+		</sockets:send-and-receive>
+		<logger level="INFO" doc:name="Logger" message='#[payload]'/>
+		<file:write path="musicians.json" config-ref="File_Config" mode="CREATE_NEW" />
+	</flow>
+----
+
+To test this example, use the same database setup as used for the sending example, run the Mule application, access the endpoint using the following netcat command `nc localhost 8085`, and send the data `{"id":2}`.
+
+== Create a Custom TCP Protocol
+
+The following example creates a TCP-based custom protocol that appends a string header to every message sent, and limits input message size to a fixed length by truncating all additional characters.
```

Comment:
```suggestion
The following example creates a TCP-based custom protocol that appends a string header to every message sent and limits input message size to a fixed length by truncating all additional characters.
```
Is this a common scenario?
Probably a silly question; I just kind of wonder how examples are chosen and if we should specify under what circumstances a user might encounter the need for this particular example.